source/dem/particle_wall_linear_force.cc

```diff
@@ -214,6 +215,187 @@ ParticleWallLinearForce<dim>::calculate_particle_wall_contact_force(
   this->mpi_correction_over_calculation_of_forces_and_torques();
 }
 
+
+template <int dim>
+void
+  ParticleWallLinearForce<dim>::calculate_particle_floating_wall_contact_force(
+    std::vector<
+      std::map<typename Triangulation<dim - 1, dim>::active_cell_iterator,
+               std::unordered_map<types::particle_index,
+                                  particle_wall_contact_info_struct<dim>>,
+               dem_data_containers::cut_cell_comparison<dim>>>
+      &                        particle_floating_mesh_in_contact,
+    const double &             dt,
+    std::vector<Tensor<1, 3>> &torque,
+    std::vector<Tensor<1, 3>> &force,
+    const std::map<unsigned int, Tensor<1, 3>>
+      &floating_mesh_translational_velocity,
+    const std::map<unsigned int, Tensor<1, 3>>
+      &                                     floating_mesh_rotational_velocity,
+    const std::map<unsigned int, Point<3>> &floating_mesh_center_of_rotation)
+{
+  std::vector<Particles::ParticleIterator<dim>> particle_locations;
+  std::vector<Point<dim>>                       triangle;
+
+  for (unsigned int solid_counter = 0;
+       solid_counter < particle_floating_mesh_in_contact.size();
+       ++solid_counter)
+    {
+      auto &particle_floating_mesh_contact_pair =
+        particle_floating_mesh_in_contact[solid_counter];
+
+      for (auto &[cut_cell, map_info] : particle_floating_mesh_contact_pair)
+        {
+          if (!map_info.empty())
+            {
+              // Get translational and rotational velocities and center of
+              // rotation
+              Tensor<1, 3> translational_velocity =
+                floating_mesh_translational_velocity.at(solid_counter);
+              Tensor<1, 3> rotational_velocity =
+                floating_mesh_rotational_velocity.at(solid_counter);
+              Point<3> center_of_rotation =
+                floating_mesh_center_of_rotation.at(solid_counter);
+
+              // Clear the particle locations vector for the new cut cell
+              particle_locations.clear();
+              const unsigned int n_particles = map_info.size();
+
+              // Gather all the particles locations in a vector
+              for (auto &&contact_info : map_info | boost::adaptors::map_values)
+                {
+                  particle_locations.push_back(contact_info.particle);
+                }
+
+              // Get cut cell id
+              // const unsigned int cut_cell_id = cut_cell_key;
+
+              // Build triangle vector
+              triangle.clear();
+              for (unsigned int vertex = 0;
+                   vertex < this->vertices_per_triangle;
+                   ++vertex)
+                {
+                  // Find vertex-floating wall distance
+                  triangle.push_back(cut_cell->vertex(vertex));
+                }
```

Comment:
Instead of clearing and push back, why not just size it the right size originally and then just add the vertex to 0 , 1 , 2? This would prevent dynamic memory allocation

Reply:
Done.